docs/source/jit.rst

```diff
@@ -471,6 +470,31 @@ Example (a type mismatch)
      ...
 
 
+Unsupported Types
+^^^^^^^^^^^^^^^^^
+Some parts of the :mod:`typing` module are missing from TorchScript. Some of these
+are more fundamental things that are unlikely to be added in the future while others
+may be added if there is enough user demand to make it a priority.
+
+These types and features from the :mod:`typing` module are unavailble in TorchScript.
+
+.. csv-table::
+   :header: "Item", "Description"
+
+   ":any:`typing.Any`", ":any:`typing.Any` is currently in development but not yet released"
+   ":any:`typing.NoReturn`", "Not implemented"
+   ":any:`typing.Union`", "Unlikely to be implemented"
+   ":any:`typing.Callable`", "Not implemented"
+   ":any:`typing.Literal`", "Not implemented"
+   ":any:`typing.ClassVar`", "Not implemented"
+   ":any:`typing.Final`", "This is supported for :any:`module attributes <Module Attributes>` class attribute annotations but not for functions"
+   ":any:`typing.AnyStr`", "TorchScript does not support :any:`bytes` so this type is not used"
+   "Type aliases", "Unlikely to be implemented"
```

Comment:
I don't think this is unlikely to be implemented. It already works when you get annotations through annotations.py.